### DISCUSSION TITLE
Migrating to new host (backup/restore)

### DISCUSSION THREAD

**Discussion by yaroz** [rustdesk/rustdesk-server-pro#184] [General]
Is there an easy way to migrate license/hosts to a new server, with minimal downtime?  Right now everything is on a local server, and all of the clients are pointed to an IP rather than a name.  We're migrating to a new cloud hosted instance.

**Comment by rustdesk** [comment]
Copy all data files (`db.sqlite3`, `db.sqlite3-shm`, `db.sqlite3-wal`, `id_ed25519`, `geo.txt` etc) under data dir on your local server to the same dir on your new server and then restart the new server program. You need also [migrate license](https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/license/#invoices-license-retrieval-and-migration) to new server.

- `db.sqlite3`, `db.sqlite3-shm`, `db.sqlite3-wal` database files, your accounts (including password), logs, settings are all stored in database
- `id_ed25519`, the `private key` is stored in this file. This file is important, it is used on `RustDesk` server, you can not recover public key without this key. The key (stored in `id_ed25519.pub`) you set on `RustDesk` client is the `public key`. You can also manually set the `private key` (the content of `id_ed25519`) on web console. 
![image](https://github.com/user-attachments/assets/590aaea5-d69d-4db7-8373-0a3dcccc32e1)
![image](https://github.com/user-attachments/assets/d6a3862b-9ac3-4f49-97c2-f0f9840b37bf)

**Reply by rustdesk** [reply]
# Where is the data (working) directory?

## Docker

It depends on how you run with docker.

Please check https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/installscript/docker/ to see where you store data files.

In our manual, we use `./data:/root` volume mapping, your data locates under local `./data` directory where you start the docker container.

## Linux debian package or `install.sh`

The data files locate at `/var/lib/rustdesk-server/`

## `RustDeskServer.setup.exe` (Deprecated)

The data files are under `C:\Program Files\RustDeskServer\bin`, but the db files are under `C:\WINDOWS\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\`.

**Reply by rustdesk** [reply]
# Below may be better for real time sqlite3 database backup/restore

## backup database
`sqlite3 /var/lib/rustdesk-server/db.sqlite3 .dump > <backup-path>/db_backup_file.sql`

## restore database
`sqlite3 /var/lib/rustdesk-server/db.sqlite3 < <backup-path>/db_backup_file.sql`

**Reply by rustdesk** [reply]
# Best practices on failover/redundant hosting

For example you can host Rustdesk server Pro in AWS using EFS for the data files.  Then having 2 instances either via load balancer or failover DNS to fail to a backup instance. 

If the two servers share the same public source ip, you can share the same license, otherwise you have to migrate license manually by yourself, https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/license/#invoices-license-retrieval-and-migration

Even without a valid license—it does not mean your failover server will not work—we still allow a 50% chance of successfully connecting to the remote device.

**Reply by rustdesk** [reply]
# database disk image is malformed.

https://github.com/rustdesk/rustdesk/issues/13489#issuecomment-3519635126

**Comment by rustdesk** [comment]
For upgrade from open source version to Pro via docker, please keep the private key `id_ed25519` file only, you can also set the private key (the content of `id_ed25519` file) manually as below screenshot (web console of the Pro). You can not reuse `db.sqlite3`, `db.sqlite3-shm`, `db.sqlite3-wal` because of compatibility issue, please do not copy them.

![image](https://github.com/rustdesk/rustdesk-server-pro/assets/71636191/48ee3495-1c50-46e7-bce9-614075aeff4a)

**Comment by ghost** [comment]
All working now.

**Comment by FCSTechAdmin** [comment]
I am not comfortable with the upgrade instructions provided. We run our RustDesk server via Docker on a Ubuntu server. Everything is working great. Upgrading without confidence is nerve racking. Is there a better step-by-step set of instructions out there?

**Reply by remoover** [reply]
same for me,.... but for my env (without lost configs, etc)
#ls /root/docker/rustdesk-server/data/compose.yml


cd /root/docker/rustdesk-server/data/
docker compose up -d --build --pull always

done.

**Reply by open-trade** [reply]
The working directory is `/root/docker/rustdesk-server`. You should `cd /root/docker/rustdesk-server`.

Supposing your initial set up is like this.
```
mkdir -p $HOME/docker/rustdesk-server
cd $HOME/docker/rustdesk-server
wget rustdesk.com/pro.yml -O compose.yml
docker compose up -d
```

Next time you upgrade, you should do
```
cd $HOME/docker/rustdesk-server
docker compose up -d --build --pull always
```

**Reply by FCSTechAdmin** [reply]
I was able to upgrade and didn't lose any data. It wasn't that bad after all.

**Comment by cleudsoncosta** [comment]
And for those who have a simple installation without using Docker? My installation is Debian 12, how can I make a backup?

**Reply by rustdesk** [reply]
[Above](https://github.com/rustdesk/rustdesk-server-pro/discussions/184#discussioncomment-8013375) mentioned it.
![image](https://github.com/user-attachments/assets/8a9cc92a-da33-4175-a2a9-fc366d3ede54)

**Comment by retechp** [comment]
Still learning the command line.  It would be great if there was an option in the web console to automatically back and/or restore system config files :)

**Reply by druneau** [reply]
Yes!  A button we can press to download a zip file, similar to how Home Assistant project does it.  Then easy restore from same zip file.

**Reply by rustdesk** [reply]
@21pages
